### PR TITLE
tuxguitar: 1.5.5 -> 1.6.3

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Description of changes

I couldn't test it on a GUI system yet.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by anthonyroussel** [comment]
Result of `nixpkgs-review pr 329133` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tuxguitar</li>
  </ul>
</details>

tuxguitar builds fine, but the binary doesn't run correctly.
When I run `./bin/tuxguitar`, tuxguitar splash screen doesn't open.

I have only a few warning lines in the logs: https://pastebin.com/LYs3EF85

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 1]
```diff
@@ -1,17 +1,12 @@
 { lib, stdenv, fetchurl, swt, jre, makeWrapper, alsa-lib, jack2, fluidsynth, libpulseaudio }:
 
-let metadata = assert stdenv.hostPlatform.system == "i686-linux" || stdenv.hostPlatform.system == "x86_64-linux";
-  if stdenv.hostPlatform.system == "i686-linux" then
-    { arch = "x86"; sha256 = "sha256-k4FQrt72VNb5FdYMzxskcVhKlvx8MZelUlLCItxDB7c="; }
-  else
-    { arch = "x86_64"; sha256 = "sha256-mj5wVQlY2xFzdulvMdb5Qb5HGwr7RElzIkpOLjaAfGA="; };
-in stdenv.mkDerivation rec {
-  version = "1.5.5";
+stdenv.mkDerivation rec {
```
Comment (anthonyroussel) [inline pkgs/applications/editors/music/tuxguitar/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

I think we could use `stdenvNoCC` here. What do you think?

**Comment by SuperSandro2000** [comment]
> tuxguitar builds fine, but the binary doesn't run correctly.
> When I run `./bin/tuxguitar`, tuxguitar splash screen doesn't open.

I've opted into using the objects from the release tarball. Our swt is severely out of date and still wants gtk2 but upstream uses not the newest version either. Now also autoPatchelfHook finds all libraries we need to run it.

File: pkgs/applications/editors/music/tuxguitar/default.nix [hunk 2]
```diff
@@ -39,7 +55,7 @@ in stdenv.mkDerivation rec {
       TuxGuitar is a multitrack guitar tablature editor and player written
       in Java-SWT. It can open GuitarPro, PowerTab and TablEdit files.
     '';
-    homepage = "http://www.tuxguitar.com.ar/";
+    homepage = "https://github.com/helge17/tuxguitar";
     sourceProvenance = with sourceTypes; [ binaryBytecode ];
```
Comment (alyssais) [inline pkgs/applications/editors/music/tuxguitar/default.nix:59]: Does this need updated?

**Comment by wegank** [comment]
Superseded by #347022.

**Pull request closed without merging**